From a3b3eea78903fd5c8f8f5ea548327851a9ae0b4c Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Tue, 5 Mar 2024 18:24:30 +0100
Subject: [PATCH 534/954] feat(st-rif): add STM32MP2x support in RIFSC driver

Adapt RIFSC driver to support all STM32MP2 variants, with specific
RIFSC IDs- in semaphore list, and maximum ID.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 64fb4e62c8c0cf4b9b3413cc457c41aa7e771f0d feat(st-rif): add STM32MP2x support in RIFSC driver]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: Ice244c4518c0b657d625bd7269ea6ad9a04e80c6
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/364527
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/st/rif/stm32_rifsc.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/drivers/st/rif/stm32_rifsc.c b/drivers/st/rif/stm32_rifsc.c
index 04dc0fbb00..8f0463b7ff 100644
--- a/drivers/st/rif/stm32_rifsc.c
+++ b/drivers/st/rif/stm32_rifsc.c
@@ -17,9 +17,21 @@
 #include <platform_def.h>
 
 static unsigned long rifsc_semaphores[] = {
+#if STM32MP21
+	STM32MP21_RIFSC_RNG2_ID,
+	STM32MP21_RIFSC_PKA_ID,
+	STM32MP21_RIFSC_SAES_ID,
+#endif /* STM32MP21 */
+#if STM32MP23
+	STM32MP23_RIFSC_RNG_ID,
+	STM32MP23_RIFSC_PKA_ID,
+	STM32MP23_RIFSC_SAES_ID,
+#endif /* STM32MP23 */
+#if STM32MP25
 	STM32MP25_RIFSC_RNG_ID,
 	STM32MP25_RIFSC_PKA_ID,
 	STM32MP25_RIFSC_SAES_ID,
+#endif /* STM32MP25 */
 };
 
 int stm32_rifsc_semaphore_init(void)
@@ -61,7 +73,15 @@ void stm32_rifsc_ip_configure(int rimu_id, int rifsc_id, uint32_t param)
 {
 	uint32_t bit;
 
+#if STM32MP21
+	assert(rifsc_id < STM32MP21_RIFSC_MAX_ID);
+#endif /* STM32MP21 */
+#if STM32MP23
+	assert(rifsc_id < STM32MP23_RIFSC_MAX_ID);
+#endif /* STM32MP23 */
+#if STM32MP25
 	assert(rifsc_id < STM32MP25_RIFSC_MAX_ID);
+#endif /* STM32MP25 */
 
 	bit = BIT(rifsc_id / U(32));
 
-- 
2.53.0

